### DISCUSSION TITLE
Retry on responseError?

### DISCUSSION THREAD

**Discussion by tobymurray** [bigskysoftware/htmx#1746] [Q&A] [answered]
On error, I want to do some client side work and re-try exactly the same request.

Presently, I have code that looks like:
```
function handleResponseError(event) {
    ...
    // Now that error has been dealt with, re-issue the event
    htmx.trigger(event.target, 'click');
}

document.body.addEventListener('htmx:responseError', handleResponseError);
```

This works when the source of the event was a 'click", but it seems to fail if the source of the event was a form `submit`. Is there a better way to re-trigger the original trigger? Or can I dynamically determine what the type of the trigger was in an event?

**Comment by Telroshan** [comment, marked as answer]
hey, the [htmx:responseError](https://htmx.org/events/#htmx:responseError) event contains a reference to the original event that triggered the request, in its `requestConfig` property.
See [this JSFiddle](https://jsfiddle.net/3ot5sqm7/1/) ; open the dev console, and notice how request are retried every 1 s, both for the button (click event) and the form (submit event), handled in a generic manner
The JS code for reference:
```html
<script type="text/javascript">
  document.body.addEventListener("htmx:responseError", function(event) {
    // Process the error

    // Retrieve which event fired the request, trigger it again
    const eventType = event.detail.requestConfig.triggeringEvent.type
    setTimeout(function() {
      htmx.trigger(event.detail.elt, eventType)
    }, 1000) // Adjust the delay as you like
  })

</script>
```

**Reply by tobymurray** [reply]
This works for me! I was using `event.target` which was fine for my application but technically incorrect, `event.detail.elt` is definitely the correct option.

I poked around a bunch through the error object and just didn't manage to find `triggeringEvent`, so much appreciated!
